modules/codemod/lib/v5/renameButtonRefs.ts

```diff
@@ -1,21 +1,44 @@
 import {API, FileInfo, Options, JSXIdentifier} from 'jscodeshift';
 import {getImportRenameMap} from './getImportRenameMap';
 
+const buttonNames = [
+  'Button',
+  'DeleteButton',
+  'deprecated_Button',
+  'DropdownButton',
+  'HighlightButton',
+  'Hyperlink',
+  'IconButton',
+  'OutlineButton',
+  'TextButton',
+  'ToolbarDropdownButton',
+  'ToolbarIconButton',
+];
+
 export default function transformer(file: FileInfo, api: API, options: Options) {
   const j = api.jscodeshift;
 
   const root = j(file.source);
 
-  const {containsCanvasImports} = getImportRenameMap(j, root, '@workday/canvas-kit-react/button');
+  const {containsCanvasImports, importMap} = getImportRenameMap(
+    j,
+    root,
+    '@workday/canvas-kit-react/button'
+  );
 
   if (!containsCanvasImports) {
     return file.source;
   }
 
   root
     .find(j.JSXIdentifier, (value: JSXIdentifier) => value.name === 'buttonRef')
-    .replaceWith(() => {
-      return j.jsxIdentifier('ref');
+    .replaceWith(nodePath => {
+      // only replace via one of our Button imports
+      const elementName = nodePath.parent?.parent?.value?.name?.name;
+      if (buttonNames.includes(elementName) || buttonNames.includes(importMap[elementName])) {
+        return j.jsxIdentifier('ref');
+      }
+      return nodePath.value;
```

Comment:
Ok, the logic here is basically:
• If a file is importing from `/button`
• and they're using one of the CK Buttons listed above
• and there's a `buttonRef` prop
• transform it to `ref`

Cool. That covers the edge cases pretty well, I think.